internal/cli/root/atlas/builder.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package atlas
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/mongodb/mongocli/internal/cli/alerts"
+	"github.com/mongodb/mongocli/internal/cli/atlas/accesslists"
+	"github.com/mongodb/mongocli/internal/cli/atlas/accesslogs"
+	"github.com/mongodb/mongocli/internal/cli/atlas/backup"
+	"github.com/mongodb/mongocli/internal/cli/atlas/cloudproviders"
+	"github.com/mongodb/mongocli/internal/cli/atlas/clusters"
+	"github.com/mongodb/mongocli/internal/cli/atlas/customdbroles"
+	"github.com/mongodb/mongocli/internal/cli/atlas/customdns"
+	"github.com/mongodb/mongocli/internal/cli/atlas/datalake"
+	"github.com/mongodb/mongocli/internal/cli/atlas/dbusers"
+	"github.com/mongodb/mongocli/internal/cli/atlas/integrations"
+	"github.com/mongodb/mongocli/internal/cli/atlas/livemigrations"
+	"github.com/mongodb/mongocli/internal/cli/atlas/logs"
+	"github.com/mongodb/mongocli/internal/cli/atlas/maintenance"
+	"github.com/mongodb/mongocli/internal/cli/atlas/metrics"
+	"github.com/mongodb/mongocli/internal/cli/atlas/networking"
+	"github.com/mongodb/mongocli/internal/cli/atlas/privateendpoints"
+	"github.com/mongodb/mongocli/internal/cli/atlas/processes"
+	"github.com/mongodb/mongocli/internal/cli/atlas/quickstart"
+	"github.com/mongodb/mongocli/internal/cli/atlas/security"
+	"github.com/mongodb/mongocli/internal/cli/atlas/serverless"
+	"github.com/mongodb/mongocli/internal/cli/auth"
+	cliconfig "github.com/mongodb/mongocli/internal/cli/config"
+	"github.com/mongodb/mongocli/internal/cli/events"
+	"github.com/mongodb/mongocli/internal/cli/iam/globalaccesslists"
+	"github.com/mongodb/mongocli/internal/cli/iam/globalapikeys"
+	"github.com/mongodb/mongocli/internal/cli/iam/organizations"
+	"github.com/mongodb/mongocli/internal/cli/iam/projects"
+	"github.com/mongodb/mongocli/internal/cli/iam/teams"
+	"github.com/mongodb/mongocli/internal/cli/iam/users"
+	"github.com/mongodb/mongocli/internal/cli/performanceadvisor"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/mongodb/mongocli/internal/version"
+	"github.com/spf13/cobra"
+)
+
+// Builder conditionally adds children commands as needed.
+func Builder(profile *string, argsWithoutProg []string) *cobra.Command {
```

Comment:
I think you can remove `argsWithoutProg` the original goal was to avoid loading atlas commands for ops manager as those do some API magic to set defaults and would fail for ops manager users when on limited internet envs 

And you can see from your tests there's no change now when calling subcommands

Reply:
Updated